web/src/use/usePaginatedResults.ts

```diff
@@ -63,10 +58,13 @@ export default function usePaginatedResult<T>(
 
   function setPage(newPage: number) {
     data.offset = (newPage - 1) * data.limit;
+    debouncedFetchResults();
   }
 
   function setItemsPerPage(newLimit: number) {
     data.limit = newLimit;
+    data.offset = 0;
+    debouncedFetchResults();
```

Comment:
I completely agree. I think this could use improvement but is a quick fix for before GSP. I can open an issue so we don't lose track of improving this in the future.

Reply:
My back-of-napkin math makes me think this is an easy fix. 

```suggestion
  function setItemsPerPage(newLimit: number) {
    data.limit = newLimit;
    data.offset = Math.floor(data.offset / newLimit) * newLimit;
    debouncedFetchResults();
```

Quick testing on my end looks good. Up to you if you want to make the change or delay.

Reply:
fixed in 03dcd772716c07cb5794e3c878d9438a542d0678